orttraining/orttraining/python/training/ortmodule/ortmodule.py

```diff
@@ -38,173 +35,168 @@ def _forward(self, *inputs, **kwargs):
+        """The original `torch.nn.Module` that this module wraps.
+
+        This property provides access to methods and properties on the original module.
+        """
+
+        # HuggingFace Trainer `save_model` method checks to see if the input model is a HuggingFace PreTrainedModel
+        # or if the model has an attribute called `module` which references a HuggingFace PreTrainedModel to save
+        # the entire context of the model so that it can be loaded using HuggingFace `from_pretrained` method.
+        # This `module` property enables HuggingFace Trainer to retrieve the underlying PreTrainedModel inside ORTModule
+        # to save and load a complete checkpoint
+
+        return self._torch_module.module
+
+    ################################################################################
+    # The methods below are part of torch.nn.Module API that are encapsulated through
+    # TorchModuleInterface
+    ################################################################################
+
     def _apply(self, fn):
         """Override original method to delegate execution to the flattened PyTorch user module"""
 
-        # Delegation must happen to _flattened_module since methods depend on
-        # _apply to recursively apply the internal setting changes
-        self._module_metadata.flattened_module._apply(fn)
+        self._torch_module._apply(fn)
         return self
 
     def apply(self: T, fn: Callable[['Module'], None]) -> T:
         """Override original method to delegate execution to the flattened PyTorch user module"""
 
-        # Delegation must happen to _flattened_module since methods depend on
-        # apply to recursively apply the internal setting changes
-        self._module_metadata.flattened_module.apply(fn)
+        self._torch_module.apply(fn)
         return self
 
     def _is_training(self):
-        return self.training and torch.is_grad_enabled()
+        return self._torch_module.is_training()
 
     def train(self: T, mode: bool = True) -> T:
         """Override original method to delegate execution to the flattened PyTorch user module"""
 
-        # Since _modules is empty, the task needs to be delegated to _module.flattened_module.train
-        # which will recursively update the original_module
         self.training = mode
-        self._module_metadata.flattened_module.train(mode)
+        # In a torch.nn.Module, _modules stores all dependent modules (sub-modules) of the current module.
+        # in a list so that torch.nn.Module can apply any changes to all sub-modules recursively.
+        # Although the _flattened_module and _original_module are dependent modules for ORTModule,
+        # they do not show up in _modules because they are abstracted away behind another class,
+        # TorchModule. In order to apply changes to those sub-modules, delegate the task to _torch_module
+        # which will recursively update the flattened_module and the original module.
```

Comment:
NIT: non blocking. This seems to be a comment regarding TorchModule implementation as opposed to ORTModule interface. Technically ORTModule doesnt know about the presence or not of a flattened module

Reply:
I'll move this in a subsequent PR to the TorchModule.